src/Service/Gateway.php

```diff
@@ -118,15 +133,26 @@ public function finalize(
         $order = $this->ginger->getOrder($_GET['order_id']);
         $context = $salesChannelContext->getContext();
         $paymentState = $order['status'];
+
+        if (isset($order['transactions']) && current($order['transactions'])['payment_method'] == 'bank-transfer'){
+            $payment_details = current($order['transactions'])['payment_method_details'];
+            $this->helper->saveIbanInfo(
+            $transaction->getOrderTransaction()->getId(),
+            $payment_details,
+            $this->orderRepository,
+            $context
+            );
+        } else
```

Comment:
How about getting rid of the else condition and saving it anyway?

Reply:
I use a check for the existence of such a key, implying that only bank-transfer has it